chainer/backends/_chainerx.py

```diff
@@ -17,6 +17,18 @@ def __init__(self, device):
         super(ChainerxDevice, self).__init__()
         self.device = device  # type: chainerx.Device
 
+    def __getstate__(self):
```

Comment:
I tried to implement this on `chainerx.Device` (pybind), but went in vain.
The reason is that `py::pickle`'s second function (`__setstate__`) cannot return `Device&` (causes compile error).
If `Device*` were returned instead, it led to segfault because it would be treated as a new instance (and pybind would try to free it).

Reply:
Succeeded by implementing `__reduce__` (with pybind) instead of using `py::pickle`.